src/comments.ts

```diff
@@ -0,0 +1,297 @@
+    return threads
+}
+
+async function createNewCommentThread(
+    document: vscode.TextDocument,
+    range: vscode.Range,
+    text: string
+): Promise<vscode.CommentThread> {
+    log.appendLine(`createNewCommentThread ${document.uri} ${range}`)
+    const [remoteUrl, branch, path] = await repoInfo(document.fileName)
+    if (remoteUrl === '') {
+        throw new Error('Git repository has no remote url configured')
+    }
+
+    const selection = getSelection(document, range)
+
+    const targetRepo: SourcegraphGQL.IDiscussionThreadTargetRepoInput = {
+        repositoryGitCloneURL: remoteUrl,
+        path,
+        branch,
+        selection,
+    }
+    const input: SourcegraphGQL.IDiscussionThreadCreateInput = {
+        title: text,
+        contents: text,
+        targetRepo,
+    }
+
+    const data = await mutateGraphQL(
+        gql`
+            mutation CreateThread($input: DiscussionThreadCreateInput!, $relativeRev: String!) {
+                discussions {
+                    createThread(input: $input) {
+                        ...DiscussionThreadFields
+                    }
+                }
+            }
+            ${discussionThreadFieldsFragment}
+        `,
+        { input, relativeRev: branch }
+    )
+
+    if (!data.discussions || !data.discussions.createThread) {
+        throw new Error(`Invalid GraphQL response for CreateThread`)
+    }
+
+    return discussionToCommentThread(document, range, data.discussions.createThread)
+}
+
+async function replyToCommentThread(
+    document: vscode.TextDocument,
+    range: vscode.Range,
+    thread: vscode.CommentThread,
+    text: string
+): Promise<vscode.CommentThread> {
+    const [, branch] = await repoInfo(document.fileName)
+    const data = await mutateGraphQL(
+        gql`
+            mutation AddCommentToThread($threadID: ID!, $contents: String!, $relativeRev: String!) {
+                discussions {
+                    addCommentToThread(threadID: $threadID, contents: $contents) {
+                        ...DiscussionThreadFields
+                    }
+                }
+            }
+            ${discussionThreadFieldsFragment}
+        `,
+        { threadID: thread.threadId, contents: text, relativeRev: branch }
+    )
+
+    if (!data.discussions || !data.discussions.addCommentToThread) {
+        throw new Error(`Invalid GraphQL response for AddCommentToThread`)
+    }
+
+    return discussionToCommentThread(document, range, data.discussions.addCommentToThread)
+}
+function discussionToCommentThread(
+    document: vscode.TextDocument,
+    range: vscode.Range,
+    thread: SourcegraphGQL.IDiscussionThread
+): vscode.CommentThread {
+    const comments: vscode.Comment[] = []
+    for (const comment of thread.comments.nodes) {
+        comments.push({
+            commentId: comment.id,
+            body: new vscode.MarkdownString(comment.contents),
+            userName: comment.author.displayName || comment.author.username, // only username?
```

Comment:
Yeah let's just show the username here, not the display name. You have to @mention usernames so it makes sense to display those. The only situation where display names make sense are e.g. on hover over a username (which AFAIK this VS Code API does not provide).